blueflood-core/src/integration-test/java/com/rackspacecloud/blueflood/io/EnumIOIntegrationTest.java

```diff
@@ -0,0 +1,333 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.io;
+
+import com.datastax.driver.core.ResultSetFuture;
+import com.google.common.collect.Table;
+import com.rackspacecloud.blueflood.io.astyanax.AstyanaxEnumIO;
+import com.rackspacecloud.blueflood.io.astyanax.AstyanaxReader;
+import com.rackspacecloud.blueflood.io.astyanax.AstyanaxWriter;
+import com.rackspacecloud.blueflood.io.datastax.DatastaxEnumIO;
+import com.rackspacecloud.blueflood.outputs.formats.MetricData;
+import com.rackspacecloud.blueflood.rollup.Granularity;
+import com.rackspacecloud.blueflood.types.*;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+/**
+ * Integration tests covering EnumIO
+ */
+public class EnumIOIntegrationTest extends IntegrationTestBase  {
+
+    protected Map<Locator, List<IMetric>> locatorToMetrics;
+    protected DatastaxEnumIO datastaxEnumIO = new DatastaxEnumIO();
+    protected AstyanaxEnumIO astyanaxEnumIO = new AstyanaxEnumIO();
+
+    /**
+     * This method is to supply the granularity parameter to some test methods below
+     *
+     * @return
+     */
+    protected Object getGranularitiesToTest() {
+        return new Object[] {
+                Granularity.FULL,
+                Granularity.MIN_5,
+                Granularity.MIN_20,
+                Granularity.MIN_60,
+                Granularity.MIN_240,
+                Granularity.MIN_1440
+        };
+    }
+
+
+    @Before
+    public void generateEnum() throws Exception {
+        locatorToMetrics = generateEnumForTenants();
+    }
+
+    @RunWith(JUnitParamsRunner.class)
+    public static class WriteDatastaxReadAstyanaxReader extends EnumIOIntegrationTest {
+
+        /**
+         * Writes enums using Datastax.
+         * <p/>
+         * Get the locator -> enum string mapping using Astyanax
+         * (i.e: call astyanaxReader.getEnumStringMappings()).
+         * <p/>
+         * Verify that the metrics_enum CF is written properly
+         *
+         * @throws Exception
+         */
+        @Test
+        public void shouldExistEnumHashToValueMapping() throws Exception {
+
+            // write using Datastax
+            for (Map.Entry<Locator, List<IMetric>> entry : locatorToMetrics.entrySet()) {
+                Locator locator = entry.getKey();
+                List<IMetric> metrics = entry.getValue();
+                for (IMetric metric : metrics) {
+                    datastaxEnumIO.putAsync(locator, metric.getCollectionTime(), (BluefloodEnumRollup) metric.getMetricValue(), Granularity.FULL);
```

Comment:
I think this one was a miss. I will add future.get().all(). 

The sleep below was needed because I wasn't getting the results. But I will re-test after I add the future().get().all and see if I still need the sleep. If not, I will remove.